src/datasets/mod.rs

```diff
@@ -0,0 +1,195 @@
+use std::fmt::Debug;
+
+use rulinalg::matrix::Matrix;
+use rulinalg::vector::Vector;
+
+/// Dataset container
+#[derive(Clone, Debug)]
+pub struct Dataset<D, T> where D: Clone + Debug, T: Clone + Debug {
+
+    data: D,
+    target: T
+}
+
+impl<D, T> Dataset<D, T> where D: Clone + Debug, T: Clone + Debug {
+
+    /// Returns explanatory variable (features)
+    pub fn data(&self) -> &D {
+        &self.data
+    }
+
+    /// Return objective variable (target)
+    pub fn target(&self) -> &T {
+        &self.target
+    }
+}
+
+
+/// Load iris dataset.
+///
+/// The data set contains 3 classes of 50 instances each, where each class refers to a type of iris plant.
+///
+/// Lichman, M. (2013). UCI Machine Learning Repository [http://archive.ics.uci.edu/ml].
+/// Irvine, CA: University of California, School of Information and Computer Science.
+pub fn load_iris() -> Dataset<Matrix<f64>, Vector<usize>> {
+    let data: Vec<f64> = vec![5.1, 3.5, 1.4, 0.2,
```

Comment:
Minor: It might be easier to use the `matrix!` macro here. My thinking is that if we need to add a row there's a little less work.